core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorUtil.java

```diff
@@ -836,7 +836,7 @@ private static ImmutableBitSet analyzeGroupExpr(SqlValidatorScope scope,
       SqlIdentifier expr = (SqlIdentifier) expandedGroupExpr;
 
       // column references should be fully qualified.
-      assert expr.names.size() == 2;
+      assert expr.names.size() >= 2;
       String originalRelName = expr.names.get(0);
       String originalFieldName = expr.names.get(1);
 
```

Comment:
Maybe we should check whether the first name is the name of a table alias in scope.

Reply:
Thanks @chunweilei for such a responsive review! 

I considered it and did a research on it but decided not to do it. The reason is that 
1. The validation is already there in this function: 
https://github.com/apache/calcite/blob/9bce62c6b6abd13d551d0eed1f1e14b024bbb2ae/core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorUtil.java#L843-L850

2. And most importantly, a better validatio is already done here: 
https://github.com/apache/calcite/blob/9bce62c6b6abd13d551d0eed1f1e14b024bbb2ae/core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java#L3946

Sorry that I should've included those reasons here.

Reply:
Makes sense.  Could you please add a negative test?

Reply:
@chunweilei Good call. Done.